### PR TITLE
Atari wallet fees

### PR DESCRIPTION
**Pull request by Quexington**
Fees already existed on the creation of a data layer but not on its update or report.

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -1387,11 +1387,14 @@ async def dl_update_root(self, request) -> Dict:
 
         for _, wallet in self.service.wallet_state_manager.wallets.items():
```
Comment (emlowe) [inline chia/rpc/wallet_rpc_api.py:1388]: How about using `.values()` and eliminate the `_`

File: chia/rpc/wallet_rpc_api.py [hunk 2]
```diff
@@ -1387,11 +1387,14 @@ async def dl_update_root(self, request) -> Dict:
...
+                records = await wallet.create_update_state_spend(
+                    bytes32.from_hexstr(request["launcher_id"]),
+                    bytes32.from_hexstr(request["new_root"]),
+                    fee=uint64(request.get("fee", 0)),
                 )
-                await self.service.wallet_state_manager.add_pending_transaction(record)
-                return {"tx_record": record.to_json_dict_convenience(self.service.config)}
+                for record in records:
+                    await self.service.wallet_state_manager.add_pending_transaction(record)
+                return {"tx_record": records[0].to_json_dict_convenience(self.service.config)}
```
Comment (emlowe) [inline chia/rpc/wallet_rpc_api.py:1397]: So we only return the "dl_tx" or the first item in this list but we call add_pending_transaction for all the records.

File: chia/data_layer/data_layer_wallet.py [hunk 3]
```diff
@@ -542,6 +578,14 @@ async def create_report_spend(self, launcher_id: bytes32) -> Tuple[TransactionRe
             type=uint32(TransactionType.OUTGOING_TX.value),
             name=singleton_record.coin_id,
         )
+        if fee > 0:
+            chia_tx = await self.create_tandem_xch_tx(fee, expected_announcement, coin_announcement=False)
+            aggregate_bundle = SpendBundle.aggregate([dl_tx.spend_bundle, chia_tx.spend_bundle])
+            dl_tx = dataclasses.replace(dl_tx, spend_bundle=aggregate_bundle)
+            chia_tx = dataclasses.replace(chia_tx, spend_bundle=None)
+            txs: List[TransactionRecord] = [dl_tx, chia_tx]
```
Comment (emlowe) [inline chia/data_layer/data_layer_wallet.py:586]: This List is always either one item (dl_tx) or two items (dl_tx, chia_tx) , correct?

File: chia/rpc/wallet_rpc_api.py (hunk 1 above)
Comment (Quexington) [reply chia/rpc/wallet_rpc_api.py:1388]: Yes I keep forgetting I can do this :)

File: chia/rpc/wallet_rpc_api.py (hunk 2 above)
Comment (Quexington) [reply chia/rpc/wallet_rpc_api.py:1397]: Yeah I didn't want to break the RPCs that were currently in use.  I don't think which tx record you get is that important, but we can fix it later if it is.

File: chia/data_layer/data_layer_wallet.py (hunk 3 above)
Comment (Quexington) [reply chia/data_layer/data_layer_wallet.py:586]: Correct, but the reason it's a list is to keep consistentcy with the equivalent wallet methods `generate_signed_transaction` which should be returning lists of tx records (CAT wallet does right now, Standard Wallet does not).

**Review by emlowe — DISMISSED** [review]
ok

**Comment by altendky** [comment]
Kick CI

**Review by emlowe — DISMISSED** [review]
Ok for me

File: chia/rpc/data_layer_rpc_api.py [hunk 4]
```diff
@@ -36,11 +37,18 @@ def process_change(change: Dict[str, Any]) -> Dict[str, Any]:
     }
 
 
+def get_fee(config: Dict[str, Any], request: Dict[str, Any]) -> uint64:
+    fee = request.get("fee")
+    if fee is None:
+        config_fee = config.get("fee", 0)
+        return uint64(config_fee)
```
Comment (altendky) [inline chia/rpc/data_layer_rpc_api.py:44]: This doesn't cover the case where the config has `None` for example from a line like `fee:` with no value I believe.

https://github.com/Chia-Network/chia-blockchain/pull/10042#discussion_r798788659

**Review by altendky — APPROVED** [review]
After a very quick scan I saw one minor hinting error and a spot where we don't handle a `None` from the config.  Maybe you could follow up in a separate PR for those.

**Pull request merged**